File: frontend/services/user-preference.ts
import { api } from './api'

export interface UserPreference {
  id: string
  user_id: string
  theme: 'Dark' | 'Light' | 'System'
  accent_color: string
  keyboard_shortcuts_enabled: boolean
  email_notifications: boolean
  in_app_notifications: boolean
  language: string
}

export const userPreferenceService = {
  /**
   * Fetch current authenticated user's UI & system settings.
   */
  async getPreferences(): Promise<UserPreference> {
    const response = await api.get<UserPreference>('/users/me/preferences')
    return response.data
  },

  /**
   * Modify UI / system preference configurations.
   */
  async updatePreferences(data: {
    theme?: 'Dark' | 'Light' | 'System'
    accent_color?: string
    keyboard_shortcuts_enabled?: boolean
    email_notifications?: boolean
    in_app_notifications?: boolean
    language?: string
  }): Promise<UserPreference> {
    const response = await api.put<UserPreference>('/users/me/preferences', data)
    return response.data
  }
}

File: backend/app/schemas/user_preference.py
import uuid
from pydantic import BaseModel
from typing import Optional

class UserPreferenceUpdate(BaseModel):
    theme: Optional[str] = None # 'Dark', 'Light', 'System'
    accent_color: Optional[str] = None
    keyboard_shortcuts_enabled: Optional[bool] = None
    email_notifications: Optional[bool] = None
    in_app_notifications: Optional[bool] = None
    language: Optional[str] = None

class UserPreferenceResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    theme: str
    accent_color: str
    keyboard_shortcuts_enabled: bool
    email_notifications: bool
    in_app_notifications: bool
    language: str

    class Config:
        from_attributes = True
